04-onion-routing.md

```diff
@@ -633,161 +633,182 @@ channel.
-which were NOT included in the onion:
 1. type: PERM|NODE|3 (`required_node_feature_missing`)
 
-A _forwarding node_ MAY, but a _final node_ MUST NOT, return one of the following
-errors:
-
-If the onion `version` byte is unknown:
 1. type: BADONION|PERM|4 (`invalid_onion_version`)
 2. data:
    * [`32`:`sha256_of_onion`]
 
-If the onion HMAC is incorrect:
 1. type: BADONION|PERM|5 (`invalid_onion_hmac`)
 2. data:
    * [`32`:`sha256_of_onion`]
 
-If the ephemeral key in the onion is unparsable:
 1. type: BADONION|PERM|6 (`invalid_onion_key`)
 2. data:
    * [`32`:`sha256_of_onion`]
 
-If an otherwise unspecified, transient error occurs in the outgoing channel
-(i.e. during forwarding to its receiving peer), e.g. channel capacity reached,
-too many in-flight HTLCs, etc.:
 1. type: UPDATE|7 (`temporary_channel_failure`)
 2. data:
    * [`2`:`len`]
    * [`len`:`channel_update`]
 
-If an otherwise unspecified, permanent error occurs during forwarding to its
-receiving peer (e.g. channel recently closed):
 1. type: PERM|8 (`permanent_channel_failure`)
 
-If the outgoing channel has requirements advertised in its
-`channel_announcement` `features` which were NOT included in the onion:
 1. type: PERM|9 (`required_channel_feature_missing`)
 
-If the receiving peer specified by the onion is NOT known:
 1. type: PERM|10 (`unknown_next_peer`)
 
-If the HTLC amount is less than the currently specified minimum amount, the
-amount of the incoming HTLC and the current channel setting for the outgoing
-channel are reported:
 1. type: UPDATE|11 (`amount_below_minimum`)
 2. data:
    * [`8`:`htlc_msat`]
    * [`2`:`len`]
    * [`len`:`channel_update`]
 
-If the HTLC does NOT pay a sufficient fee, the amount of the incoming HTLC and
-the current channel setting for the outgoing channel are reported:
 1. type: UPDATE|12 (`fee_insufficient`)
 2. data:
    * [`8`:`htlc_msat`]
    * [`2`:`len`]
    * [`len`:`channel_update`]
 
-If the `outgoing_cltv_value` does NOT match the `update_add_htlc`'s
-`cltv_expiry` minus the `cltv_expiry_delta` for the outgoing channel, the
-`cltv_expiry` and the current channel setting for the outgoing channel are
-reported:
 1. type: UPDATE|13 (`incorrect_cltv_expiry`)
 2. data:
    * [`4`:`cltv_expiry`]
    * [`2`:`len`]
    * [`len`:`channel_update`]
 
-If the `cltv_expiry` is too near, the current channel setting for the
-outgoing channel are reported:
 1. type: UPDATE|14 (`expiry_too_soon`)
 2. data:
    * [`2`:`len`]
    * [`len`:`channel_update`]
 
-If the `cltv_expiry` is unreasonably far in the future, its also possible to
-report an error:
 1. type: 21 (`expiry_too_far`)
 
-If the channel is disabled, the current channel setting for the outgoing
-channel are reported:
 1. type: UPDATE|20 (`channel_disabled`)
 2. data:
    * [`2`: `flags`]
    * [`2`:`len`]
    * [`len`:`channel_update`]
+[FIXME: is it important that 20 and 21 remain in this order or can they be moved to end of list?]
```

Comment:
Order can (and should!) be shuffled into logical ordering.  Numbers themselves are mainly relics of history, but hard to change.

Reply:
Ok, done.